File: app/src/java/java/com/example/ama/viewer/di/AppComponent.java
package com.example.ama.viewer.di;

import com.example.ama.viewer.ViewerApp;
import com.example.ama.viewer.di.modules.AppModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class})
public interface AppComponent extends AndroidInjector<ViewerApp> {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder context(ViewerApp app);

        AppComponent build();
    }
}

File: app/src/kotlin/java/com/example/ama/viewer/di/AppComponent.kt
package com.example.ama.viewer.di

import com.example.ama.viewer.ViewerApp
import com.example.ama.viewer.di.modules.AppModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    AppModule::class])
interface AppComponent : AndroidInjector<ViewerApp> {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun context(app: ViewerApp): Builder

        fun build(): AppComponent
    }
}